### PR TITLE
Feature/test map state to props

### PR DESCRIPTION
**Pull request by Volubyl**
DO NOT MERGE BEFORE : https://github.com/CoorpAcademy/components/pull/1498

## Detailed purpose of the PR

Currently, it was quite difficult to test the mapStateToProps function because mocking the progression and the state appeared to be painfull.

Now, I create two factories one to create a progression (could be improve following future needs) and a redux store.

Those functions fits quite well with the functions (createSlide, createChapter) written by @VincentCATILLON 

To see a exemple of use please refer to thise file : `src/containers/header-slide-title.test.js`

### Result and observation

Consider the `createStore` function as a V1. It covers our basic needs but could be upgrade furtherly

- [ ] **Breaking changes ?**  
       If checked, what have you broken ?

- [ ] **Extra dependency ?**
      If checked, Which extra dependency did you add ? (name, purpose, link ...).

### Testing Strategy

- [ ] Already covered by tests
- [ ] Manual testing
- [X] Unit testing

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,355 +1,355 @@
 version: 2
```
Comment (Volubyl) [inline .circleci/config.yml]: mmm no changes here except the indentation. I guess Visual studio is formatting yml differently

File: src/__fixtures__/mappers.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
+// @flow
```
Comment (VincentCATILLON) [inline src/__fixtures__/mappers.js]: Why is this in fixtures ?

File: src/__fixtures__/progression.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
+// @flow
+import type {Progression, GenericContent} from '@coorpacademy/progression-engine';
+
+type Engine = 'microlearning' | 'learner';
```
Comment (VincentCATILLON) [inline src/__fixtures__/progression.js]: Should be defined elsewhere than there

File: src/__fixtures__/store.js [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+  const mappedSlide: {[key: string]: SlideEngine} = createMapObject(slides.map(mapToSlide));
+  const mappedChapter: {[key: string]: ChapterStore} = createMapObject(chapters.map(mapToChapter));
+  const mappedDisciplines: {[key: string]: DisciplinStore} = createMapObject(
+    disciplines.map(mapToDiscipline)
+  );
+
+  const disciplineBundleState: DisciplineBundleState = {
+    chapters: {},
+    disciplines: {}
+  };
```
Comment (VincentCATILLON) [inline src/__fixtures__/store.js]: Can't you take all exported initialState instead of duplicating all ones ?

File: src/__fixtures__/store.js [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+  if (typeof currentValue._id === 'string') {
+    return {
+      ...accumulator,
+      [currentValue._id]: currentValue
+    };
+  }
+  return {
+    ...accumulator
+  };
+};
```
Comment (VincentCATILLON) [inline src/__fixtures__/store.js]: Wow, very complex, can you explain of add some comments? :|

File: .circleci/config.yml [hunk 6]
```diff
@@ -1,355 +1,355 @@
...
+            - deploy_android:
+                  requires:
+                      - deploy_prepare
+                      - build_android
+            - deploy_ios:
+                  requires:
+                      - deploy_prepare
+                      - build_ios
+                      # to prevent push fails (and build android is faster)
+                      - deploy_android
```
Comment (VincentCATILLON) [inline .circleci/config.yml]: Please revert these changes please. This will make huge conflicts

File: src/__fixtures__/store.js [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+  Chapter as ChapterStore,
+  Discipline as DisciplinStore
+} from '@coorpacademy/player-store';
+import type {Slide as SlideEngine, Progression} from '@coorpacademy/progression-engine';
+import type {Level, Slide, Chapter, Discipline} from '../layer/data/_types';
+import type {StoreState} from '../redux/store';
+import {initialState as bundledDisciplineState} from '../redux/reducers/discipline-bundle';
+import {mapToLevel, mapToSlide, mapToChapter, mapToDiscipline} from './utils/mappers';
+
+type MappableObject =
```
Comment (Fausto95) [inline src/__fixtures__/store.js]: MappableRef ou MappableID instead

File: src/__fixtures__/store.test.js [hunk 8]
```diff
@@ -0,0 +1,208 @@
+// @flow
+import {createMapObject, createStoreState} from './store';
+
+import {createLevel} from './levels';
+import {createSlide} from './slides';
+import {createQCM} from './questions';
+import {createProgression} from './progression';
+
+describe('storeFixture', () => {
+  it('should reduce an array  of Mappble Objects to an object -- with ref', () => {
```
Comment (Fausto95) [inline src/__fixtures__/store.test.js]: Mappable

File: src/__fixtures__/store.test.js [hunk 9]
```diff
@@ -0,0 +1,208 @@
...
+      [dummyObject1.ref]: dummyObject1,
+      [dummyObject2.ref]: dummyObject2
+    };
+
+    const result = createMapObject(mappableObject);
+
+    expect(result).toEqual(expectedResult);
+  });
+
+  it('should reduce an array  of Mappble Objects to an object -- with universalRef', () => {
```
Comment (Fausto95) [inline src/__fixtures__/store.test.js]: typo

File: src/__fixtures__/store.test.js [hunk 10]
```diff
@@ -0,0 +1,208 @@
...
+      [dummyObject1.universalRef]: dummyObject1,
+      [dummyObject2.universalRef]: dummyObject2
+    };
+
+    const result = createMapObject(mappableObject);
+
+    expect(result).toEqual(expectedResult);
+  });
+
+  it('should reduce an array  of Mappble Objects to an object -- with _id', () => {
```
Comment (Fausto95) [inline src/__fixtures__/store.test.js]: typo

File: src/__fixtures__/utils/mappers.js [hunk 11]
```diff
@@ -0,0 +1,94 @@
...
+
+import type {
+  Level as LevelStore,
+  Chapter as ChapterStore,
+  Discipline as DisciplineStore
+} from '@coorpacademy/player-store';
+import type {Slide as SlideEngine, Lesson as LessonEngine} from '@coorpacademy/progression-engine';
+import type {Level, Slide, Lesson, Chapter, Discipline} from '../../layer/data/_types';
+
+export const mapToLevel = (rawLevel: Level): LevelStore => ({
```
Comment (Fausto95) [inline src/__fixtures__/utils/mappers.js]: Why don't you use the existing mappers instead of creating newer ones
